frontend/js/tests/user/import-listens/ImportListens.test.tsx

```diff
@@ -0,0 +1,194 @@
+    window.HTMLElement.prototype.scrollIntoView = jest.fn();
+
+    server = setupServer(
+      http.post("/settings/import/", (req) => {
+        return HttpResponse.json({ user_has_email: true });
+      }),
+      http.get("/1/import-listens/list/", (req) => {
+        return HttpResponse.json(mockImports);
+      }),
+      http.post("/1/import-listens/cancel/1/", (req) => {
+        return HttpResponse.json({ success: true, id: 1 });
+      }),
+      http.get("/1/import-listens/1/", (req) => {
+        return HttpResponse.json(mockImports[0]);
+      })
+    );
+
+    server.events.on("request:start", ({ request }) => {
+      console.log("MSW captured:", request.method, request.url);
+    });
+
+
+    server.listen();
+
+    router = createMemoryRouter(routes, {
+      initialEntries: ["/settings/import/"],
+    });
+  });
+
+  beforeEach(() => {
+    submitAPICalled = false;
+    cancelAPICalled = false;
+    refreshAPICalled = false;
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it("renders submission form correctly", async () => {
+    renderWithProviders(<RouterProvider router={router} />, {}, { wrapper: ReactQueryWrapper }, false);
+    await waitFor(() => {
+      expect(screen.getByText(/start import from/i)).toBeInTheDocument();
+      expect(screen.getByText(/end date for import/i)).toBeInTheDocument();
+      expect(screen.getByText(/select Service/i)).toBeInTheDocument();
+      expect(screen.getByText(/choose a File/i)).toBeInTheDocument();
+      expect(screen.getByRole("button", { name: /import listens/i })).toBeInTheDocument();
+    });
+  });
+
+  it("enables the import button after a file is uploaded", async () => {
+    renderWithProviders(<RouterProvider router={router} />, {}, { wrapper: ReactQueryWrapper }, false);
+    const importButton = screen.getByRole("button", { name: /import listens/i });
+    expect(importButton).toBeDisabled();
+
+    const file = new File(['{ "foo": "bar" }'], 'test.json', { type: 'application/json' });
+    const input = screen.getByLabelText(/choose a file/i);
+
+    fireEvent.change(input, { target: { files: [file] } });
+
+    await waitFor(() => {
+      const importButtonAfter = screen.getByRole("button", { name: /import listens/i });
+      expect(importButtonAfter).not.toBeDisabled();
+    });
+  });
+
+  it("renders imports with correct data", async () => {
+    renderWithProviders(<RouterProvider router={router} />, {}, { wrapper: ReactQueryWrapper }, false);
+    await waitFor(() => {
+      expect(screen.getByText("spotify.zip")).toBeInTheDocument();
+      expect(screen.getByText("August 11th, 2025")).toBeInTheDocument();
+      expect(screen.getByText("-")).toBeInTheDocument();
+    });
+  });
+
+it("calls import endpoint when import listens clicked", async () => {
+  let submitAPICalled = false;
+
+  server.use(
+    http.post("/settings/import/", (req) => {
+      submitAPICalled = true;
+      return HttpResponse.json({ user_has_email: true });
```

Comment:
Shouldn't return this content.

While testing locally I created a new mock import:
```ts
const newImport = {
  import_id: 3,
  service: "spotify",
  created: "2025-08-12T10:00:00Z",
  metadata: {
    filename: "spotify_test.zip",
    progress: "Your data import will start soon.",
    status: "waiting",
  },
  from_date: "1970-01-01T00:00:00Z",
  to_date: "2025-08-11T00:00:00Z",
};
```

Then use it like so:

```suggestion
      return HttpResponse.json(newImport);
```